lib/dbservice/groups.ex

```diff
@@ -31,6 +31,32 @@ defmodule Dbservice.Groups do
   """
   def get_group!(id), do: Repo.get!(Group, id)
 
+  @doc """
+  Gets group by group_id.
+  Raises `Ecto.NoResultsError` if the Group does not exist.
+  ## Examples
+      iex> get_group_by_group_id(123)
+      %Group{}
+      iex> get_group_by_group_id(456)
+      ** (Ecto.NoResultsError)
+  """
+  def get_group_by_group_id(group_id) do
+    Repo.get_by(Group, id: group_id, type: "batch")
```

Comment:
why hardcode `batch` here? maybe this can be in input too?

Reply:
Currently we always have sessions for a batch. Later if there is a need where session are associated with a program, product, etc. then we can have a more generic function where type can be just passed as an argument of this function.

Reply:
got it

Reply:
i'm making an issue so that we can remember this for future